sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -764,3 +748,146 @@ def update_repository_properties(self, *args, **kwargs):
                 repository, value=properties._to_generated(), **kwargs  # pylint: disable=protected-access
             )
         )
+
+    @distributed_trace
+    def upload_manifest(
+        self, repository: str, manifest: "Union['OCIManifest', 'IO']", *, tag: "Optional[str]" = None, **kwargs: "Any"
+    ) -> None:
+        """Upload a manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param manifest: The manifest to upload.
+        :type manifest: ~azure.containerregistry.models.OCIManifest or IO
+        :keyword tag: Tag of the manifest.
+        :paramtype tag: str or None
+        :returns: None
+        :rtype: None
+        :raises ValueError: If the parameter repository or manifest is None.
+        """
+        if (repository or manifest) is None:
+            raise ValueError("The parameter repository and manifest cannot be None.")
+
+        tag_or_digest = tag
+        if tag is None:
+            stream = manifest
+            if isinstance(manifest, OCIManifest):
+                stream = _serialize_manifest(manifest)
+            tag_or_digest = _compute_digest(stream)
+
+        self._client.container_registry.create_manifest(
+            name=repository, reference=tag_or_digest, payload=stream, content_type=OCI_MANIFEST_MEDIA_TYPE, **kwargs
+        )
+
+    @distributed_trace
+    def upload_blob(self, repository, stream, **kwargs):
```

Comment:
I'd prefer we rename the `stream` parameter to something like `data`.
What's the maximum size supported here? Or minimum size? Would I ever want to pass in bytes directly? Or will always be a stream/file handle/etc?

Reply:
* Have changed `stream` to `data`
* We haven't talked about the blob size, I didn't find there's any size check in other languages. I think that's a good question to ask in arch board review(5/10).
* We accept consuming bytes IO instead of bytes here.

Reply:
Today there is no size limitation in the SDK. We assume the bytes given to us our repayable. Our current thinking is to build an experience on re-playability and size management once we get customer asks. (As of now the only customer for the library is Bicep which does not need this support).